### ISSUE TITLE
Change Halo lib to Rich

### ISSUE THREAD

**Issue by gabrielcocenza** [canonical/charmed-openstack-upgrader#382]
Halo it's not supported anymore and it's not possible to have multiple indicators for tasks that are running in parallel. COU should eventually change to something that is able to show multiple indicators like the rich python lib

**Comment by aieri** [comment]
this could be useful to improve debuggability thanks to the tracebacks showing local variables.

**Comment by Pjack** [comment]
Close this JIRA ticket. 
Refactoring has become a low priority, as Caracal will be the final release of charmed-openstack.
